mlir/tools/mlir-miopen-lib/mlir-miopen-lib.cpp

```diff
@@ -124,51 +129,102 @@ extern "C" MlirHandle CreateMlirHandle(const char *arguments) {
         strToInt("padding_h"), strToInt("padding_w"), module, builder,
         argMap["kernel_name"], mlir::FloatType::getF32(&(handle->context)),
         false);
-
-    if (argMap["arch"] != "")
-      s_arch = argMap["arch"];
   }
 
   return handle;
 }
 
-extern "C" void MlirLowerCpp(MlirHandle mlirHandle) {
+extern "C" int DestroyMlirHandle(MlirHandle mlirHandle) {
   MlirHandle_s *handle = static_cast<MlirHandle_s *>(mlirHandle);
+  if (handle == nullptr)
+    return EMlirInvalidParam;
+
+  delete handle;
+  return EMlirSuccess;
+}
+
+extern "C" int MlirGetExecutionDims(MlirHandle mlirHandle, size_t *global_size,
+                                    size_t *local_size) {
```

Comment:
Nice job.

I was pondering between one of the behaviors:
  - Client pass in the block and grid size from the handle creation and we are forced to use passed in block and grid size
  - MLIR return the block/grid size, as you did
I do feel like the way you do it make more sense.

Reply:
I think we only allow that for testing different block_size. From a producer perspective we should know what the best block size is.

Reply:
Agreed, unfortunately MIOpen try to make its own decisions and I need a way to override ours with theirs. TBD for me.